src/client/components/EditUserForm/CollaboratorPermissions/CollaboratorPermissions.tsx

```diff
@@ -33,52 +37,40 @@ const CollaboratorPermissions = (props: Props) => {
 
   if (!sections) return null
 
-  const options = SubSections.getAnchorsByUuid({ cycle, sections })
-
   const permissions = (userRole.permissions as CollabPermissions) || undefined
 
-  const sectionPermissions = Objects.isEmpty(permissions) ? 'all' : permissions.sections ?? 'none'
-
-  const tableDataPermissions = Object.entries(sectionPermissions)
-    .reduce((prev, [k, v]) => (v.tableData ? [...prev, options[k]] : prev), [])
-    .sort()
-    .join(', ')
-
-  const descriptionsPermissions = Object.entries(sectionPermissions)
-    .reduce((prev, [k, v]) => (v.descriptions ? [...prev, options[k]] : prev), [])
-    .sort()
-    .join(', ')
+  const handlePermissionsChange = (permissions: CollabPermissions) => {
+    dispatch(
+      UserManagementActions.updateSectionAuth({
+        id: userRole.id,
+        sections: permissions?.sections,
+        params: {
+          assessmentName,
+          cycleName,
+          countryIso,
+        },
+      })
+    )
+  }
 
   return (
     <div className="edit-user__form-item edit-user__form-item-permissions">
       <div className="edit-user__form-label">{t(`userManagement.permissions`)}</div>
 
       <div className="edit-user__form-field edit-user__form-field-premissions">
-        <div className="edit-user__form-field-premission-list">
-          {typeof sectionPermissions === 'string' ? (
-            t(`contactPersons.${sectionPermissions}`)
-          ) : (
-            <>
-              {tableDataPermissions !== '' && (
-                <div>
-                  {t('userManagement.permissionNames.tableData')}: {tableDataPermissions}
-                </div>
-              )}
-              {descriptionsPermissions !== '' && (
-                <div>
-                  {t('userManagement.permissionNames.descriptions')}: {descriptionsPermissions}
-                </div>
-              )}
-            </>
-          )}
-        </div>
+        <CollaboratorAccessList permissions={permissions} />
 
         <button className="btn-xs btn-primary" onClick={_onEditPermissionsClick} type="button">
           {t('userManagement.editPermissions')}
         </button>
       </div>
 
-      <CollaboratorAccessModal open={modalOptions.open} userRole={userRole} onClose={_onEditPermissionsClose} />
+      <CollaboratorAccessModal
```

Comment:
Sounds good!

Reply:
Added `CollaboratorPermissionsEditor` to  `client/components/user` and `ModalEditor` within it. It's more organized now, thanks!

Reply:
What do you think of moving`<CollaboratorAccessList>` to be an inner component of the current `<CollaboratorPermissionsEditor>`, and then the entire editor would be:
```
CollaboratorPermissionsEditor:
    <>
      <CollaboratorAccessList/>

      <button/>

      <ModalEditor />
    </>
``` 
So that the logic of `<CollaboratorAccessList>` (generating the table and description permissions) is kept separate. 
Then I also only render `<CollaboratorPermissionsEditor>` as you suggested above.

Reply:
Added 👍 I reorganized the styles too (looks the same).